video/out/mac/common.swift

```diff
@@ -410,8 +410,9 @@ class Common: NSObject {
     }
 
     func getScreenBy(name screenName: String?) -> NSScreen? {
+        if (screenName == nil) { return nil }
         for screen in NSScreen.screens {
-            if screen.displayName == screenName {
+            if screen.localizedName == screenName {
```

Comment:
sorry fpr the long wait. i still see one problem here. when the binary is compiled with an SDK +10.15 (HAVE_MACOS_10_15_FEATURES) but with a target OS target lower than this. in this case it doesn't extend the NSScreen and implements the localizedName function, but run on an OS <10.15 it still tries to call this none available function.

Reply:
So it would be best to keep the `displayName` extension and add a compile/runtime check there to return `localizedName` if available?
This is what I came up with. Would this be sufficient/acceptable?
```swift
    public var displayName: String? {
        get {
            var name: String? = nil
            #if HAVE_MACOS_10_15_FEATURES
            if #available(macOS 10.15, *) {
                name = localizedName
            }
            else {
                name = displayNameLegacy
            }
            #else
            name = displayNameLegacy
            #endif
            return name;
        }
    }

    private var displayNameLegacy: String? { … }
```

Somehow the compiler always optimizes out the legacy part even when declaring `MACOSX_DEPLOYMENT_TARGET=10.8`, so I can't really test this. Do you have any hints?

Reply:
Maybe the following approach is better.

```swift
    public var displayName: String? {
        get {
            var name: String? = nil
            let localizedNameSelector = Selector("localizedName")
            if (NSScreen.instancesRespond(to: localizedNameSelector)) {
                name = self.perform(localizedNameSelector).takeRetainedValue() as! String
            }
            else {
                name = displayNameLegacy
            }
            return name;
        }
    }
```

Reply:
no i think the first one is the better approach but a bit too complex. i don't think you need a variable to assign the right name and return that var. just return it directly.

how about something like this (untested):
```swift
private var displayNameLegacy: String? {
    // ...
    return "..."
}

public var displayName: String? {
    get {
#if HAVE_MACOS_10_15_FEATURES
        if #available(macOS 10.15, *) {
            return localizedName
        }
#endif
        return displayNameLegacy;
    }
}
```

on a side note, the compile time checks should always be without indentation.

you probably also need to explicitly set the swift target `--swift-flags="-target x86_64-apple-macosx10.12"` too.

Reply:
Thx, it now seems to keep both branches with the meson build option ```-Dswift-flags="-target x86_64-apple-macosx10.12"```. I will update the commit accordingly.